### PR TITLE
Test operator source

### PR DESCRIPTION
**Pull request by pmacik**
This PR:

* Adds a make target (`test-operator-source`) to execute a simple test for verifying that the operator is successfully installed from application registry on Quay.io.

It is a functional test for `OperatorSource`. The files included in the pull request define the `OperatorSource` itself and it's corresponding `CatalogSource`, and `Subscription`.

It depends on the `push-operator-app-registry` make target from https://github.com/redhat-developer/devconsole-operator/pull/174 that creates a CSV for the operator and pushes it to quay.io. The `test-operator-source` make target runs the test to verify the creation of the `Subscription`, `InstallPlan`, and verifies that the operator's pod has started.

### CODE REVIEW

**Comment by pmacik** [comment]
This PR replaces and extends the #135.

**Comment by ldimaggi** [comment]
Small syntax error in make file:

Line 106 should be:
```
$(eval INSTALL_PLAN := $(shell oc get sub devconsole -n openshift-operators -o jsonpath='{.status.installplan.name}'))
```

**Comment by ldimaggi** [comment]
I'd also be interested in seeing the state of the pod - so that the test will cover these steps:

```
.PHONY: test-operator-source
test-operator-source: push-operator-app-registry
	$(eval OPSRC_NAME := devconsole-operators-$(TAG))
	$(Q)oc project openshift-marketplace 
	$(Q)sed -e "s,REPLACE_NAMESPACE,$(DEVCONSOLE_APPR_NAMESPACE)," ./test/opsrc/operator_source.yaml | sed -e "s,REPLACE_OPERATOR_SOURCE_NAME,$(OPSRC_NAME)," | oc apply -f -
	$(Q)oc apply -f ./test/opsrc/catalog_source.yaml
	$(Q)oc apply -f ./test/opsrc/subscription.yaml
	$(Q)oc get sub devconsole -n openshift-operators
	$(eval INSTALL_PLAN := $(shell oc get sub devconsole -n openshift-operators -o jsonpath='{.status.installplan.name}'))
	$(Q)oc get installplan $(INSTALL_PLAN) -n openshift-operators
	$(Q)oc get pods  -l name=devconsole-operator -n openshift-operators -o jsonpath='{.items[*].status.phase}{"\n"}'
```
We will want to wrap the actual test steps in Go so that the perform asserts.

**Comment by sbose78** [comment]
Add a description please :)

**Comment by ldimaggi** [comment]
> Add a description please :)

This pull request implements a functional test for OperatorSource. The files included in the pull request define the Operator Source, it's corresponding CatalogSource, and Subscription. 

The push-operator-app-registry make target creates the OperatorSouce and pushes the Operator to quay.io. The test-operator-source make target runs the test to verify the creation of the OperatirSubscription, install plan, and verifies that the Operator's pod has started.

**Comment by pmacik** [comment]
@sbose78, @ldimaggi Updated the PR's description.

File: test/opsrc/catalog_source.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: operators.coreos.com/v1
```
Comment (sbose78) [inline test/opsrc/catalog_source.yaml]: Name of the file should be `catalogsourceconfig.yaml`

File: test/opsrc/basic_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package opsrc
```
Comment (sbose78) [inline test/opsrc/basic_test.go]: please use `operatorsource` as the package name

**Comment by sbose78** [comment]
Nice work @pmacik and @ldimaggi !

File: test/opsrc/catalog_source.yaml (hunk 1 above)
Comment (pmacik) [reply test/opsrc/catalog_source.yaml]: Thanks @sbose78, good catch! I renamed the file.

File: test/opsrc/basic_test.go (hunk 2 above)
Comment (pmacik) [reply test/opsrc/basic_test.go]: I renamed the package.

**Pull request merged**